engine/configurator/conf_cyaml.h

```diff
@@ -0,0 +1,806 @@
+     *
+     * Its CYAML type is string pointer, and we have no minimum or maximum
+     * string length constraints.
+     */
+    CYAML_FIELD_STRING_PTR(
+                           "reboot_ta", CYAML_FLAG_POINTER | CYAML_FLAG_OPTIONAL,
+                           history_entry, reboot_ta, 0, CYAML_UNLIMITED),
+
+    /* The field array must be terminated by an entry with a NULL key.
+     * Here we use the CYAML_FIELD_END macro for that. */
+    CYAML_FIELD_END
+};
+
+/* The value for a history_entry is a mapping.
+ *
+ * Its fields are defined in history_entry_fields_schema.
+ */
+static const cyaml_schema_value_t history_entry_schema = {
+
+        CYAML_VALUE_MAPPING(CYAML_FLAG_DEFAULT,
+                            history_entry,
+                            history_entry_fields_schema),
+};
+
+/* The history field definitions schema is an array.
+ *
+ * All the field entries will refer to their parent mapping's structure,
+ * in this case, `history_seq`.
+ */
+static const cyaml_schema_field_t history_fields_schema[] = {
+    /* The only field is the register, unregister, add or something.
+     *
+     * YAML key: "history".
+     * C structure member for this key: "entries".
+     *
+     * Its CYAML type is a sequence.
+     *
+     * Since it's a sequence type, the value schema for its entries must
+     * be provided too. In this case, it's history_entry_schema.
+     *
+     * Since it's not a sequence of a fixed-length, we must tell CYAML
+     * where the sequence entry count is to be stored. In this case, it
+     * goes in the "entries_count" C structure member in
+     * history_seq.
+     * Since this is "entries" with the "_count" postfix, we can use
+     * the following macro, which assumes a postfix of "_count" in the
+     * struct member name.
+     */
+    CYAML_FIELD_SEQUENCE(
+                         "history",
+                         CYAML_FLAG_POINTER | CYAML_FLAG_OPTIONAL,
+                         history_seq, entries,
+                         &history_entry_schema, 0, CYAML_UNLIMITED),
+    /* The field array must be terminated by an entry with a NULL key.
+     * Here we use the CYAML_FIELD_END macro for that. */
+    CYAML_FIELD_END
+};
+
+
+
+
+
+/* Top-level schema of the history. The top level value for the history is
+ * a sequence.
+ *
+ * Its fields are defined in history_entry_fields_schema.
+ */
+static const cyaml_schema_value_t history_schema = {
+    CYAML_VALUE_MAPPING(CYAML_FLAG_POINTER,
+                        history_seq, history_fields_schema),
+};
+
+/* Our CYAML config.
+ *
+ * If you don't want to change it between calls, make it const.
+ *
+ * Here we have a very basic config that difined in conf_cyaml.c.
+ */
+extern cyaml_config_t cyaml_config;
+
+/**
+ * Process CYAML errors to TE errors
+ *
+ * @param err          CYAML error
+ *
+ * @return status code (see te_errno.h)
+ */
+extern te_errno
+te_process_cyaml_errors(cyaml_err_t err);
```

Comment:
`te_` is a strange prefix here. I'd say it's `cyaml_`, to match other objects in the same file. Or `cfg_` ?

Reply:
Fixed.